src/zc/buildout/tests.py

```diff
@@ -3062,6 +3062,39 @@ def test_abi_tag_eggs():
     ...
     """
 
+def test_buildout_doesnt_keep_adding_itself_to_versions():
+    r"""
+    We were constantly writing to versions.cfg for buildout and setuptools
+
+    >>> write('buildout.cfg',
+    ... '''
+    ... [buildout]
+    ... parts =
+    ... extends = versions.cfg
+    ... show-picked-versions = true
+    ... update-versions-file = versions.cfg
+    ... extends = versions.cfg
+    ... ''')
+    >>> write('versions.cfg',
+    ... '''[versions]
+    ... ''')
+    >>> _ = system(join('bin', 'buildout'))
+    >>> with open('versions.cfg') as f:
+    ...     versions = f.read()
+    >>> _ = system(join('bin', 'buildout'))
+    >>> _ = system(join('bin', 'buildout'))
+    >>> _ = system(join('bin', 'buildout'))
+    >>> with open('versions.cfg') as f:
+    ...     versions == f.read()
+    True
+    >>> cat('versions.cfg') # doctest: +ELLIPSIS
```

Comment:
The check above makes sure that versions.cfg didn't change after the first run.  The check on the 3090 sanity checks that something was added on the first run.  I'll rearrange this to make it clearer.